lib/rex/proto/rfb/constants.rb

```diff
@@ -13,39 +13,69 @@
 ##
 
 module Rex
-module Proto
-module RFB::Constants
+  module Proto
+    module RFB::Constants
+      DefaultPort = 5900
 
-DefaultPort = 5900
+      # Version information
+      MajorVersions = [3, 4, 5]
+      # NOTE: We will emulate whatever minor version the server reports.
 
-# Version information
-MajorVersions = [3, 4]
-# NOTE: We will emulate whatever minor version the server reports.
+      # Security types https://datatracker.ietf.org/doc/html/rfc6143#section-8.1.2
+      # https://www.iana.org/assignments/rfb/rfb.xhtml
+      class AuthType
+        Invalid = 0
+        None = 1
+        VNC = 2
+        # RealVNC has 3-15 registered
+        RealVNC_3 = 3
+        RealVNC_4 = 4
+        RA2 = 5
+        RA2ne = 6
+        RealVNC_7 = 7
+        RealVNC_8 = 8
+        RealVNC_9 = 9
+        RealVNC_10 = 10
+        RealVNC_11 = 11
+        RealVNC_12 = 12
+        RealVNC_13 = 13
+        RealVNC_14 = 14
+        RealVNC_15 = 15
+        Tight = 16
+        Ultra = 17
+        TLS = 18
+        VeNCrypt = 19 # TigerVNC this is Plain, TLSNone, TLSVnc, TLSPlain, X509Vnc, X509Plain
```

Comment:
Is this used when one of those options is specified? Little unclear on the relation via just the wording `this is` here? Is it, `In TigerVNC this is used when one of the following options is specified:`?

Reply:
you are correct, TigerVNC seemed to shove all those options under one auth and then it must negotiate them later somehow.  Your re-wording is correct though from what I remember.

Reply:
Will reword this and upload changes, just to make it at little clearer.

Reply:
Fixed in https://github.com/rapid7/metasploit-framework/pull/16207/commits/4e8613904ae32b9744bb8eb79afc851f341d2d14